File: top150/_01_ArrayString/q58.java
package top150._01_ArrayString;

public class q58 {
    public int lengthOfLastWord(String s) {
        int len = 0;
        char[] arr = s.toCharArray();

        int i = arr.length-1;
        while(i >= 0 && arr[i] == ' ') {
            i--;
        }

        while(i >= 0 && arr[i] != ' ') {
            len++;
            i--;
        }
    

        return len;
    }
}

File: top150/_01_ArrayString/q58.py
class Solution:
    def lengthOfLastWord(s):
        l = 0
        n = len(s)
        r = n-1
        count = 0
        while r >= 0:
            if s[r] != ' ':
                count += 1
            elif count != 0:
                return count
            r -= 1
        return count
